[Bug] Fix TriggerRelationMapper cannot work due to miss DatabaseIdProvider

mind-echo opened the pull request:
fix #15152 : When using PostgreSQL, MyBatis did not set databaseId correctly, causing the workflow to fail to start

<!--Thanks very much for contributing to Apache DolphinScheduler, we are happy that you want to help us improve DolphinScheduler! -->

## Purpose of the pull request

fix bug #15152

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

1. add bean of VendorDatabaseIdProvider
2. change org.apache.dolphinscheduler.dao.mapper.TriggerRelationMapper#upsert.databaseId from pg to PostgreSQL

## Verify this pull request

Commit:
fix bug 15152: When using PostgreSQL, MyBatis did not set databaseId correctly, causing the workflow to fail to start

fuchanghai reviewed (approved):
LGTM
BTW, do we need to do some mapping ? as following :
```
@Bean
public DatabaseIdProvider databaseIdProvider() {
    VendorDatabaseIdProvider databaseIdProvider = new VendorDatabaseIdProvider();
    Properties properties = new Properties();
    properties.put("Oracle","oracle");
    properties.put("MySQL","mysql");
    properties.put("PostgreSQL","pg");
    databaseIdProvider.setProperties(properties);
    return databaseIdProvider;
}
```
cc @EricGao888

Commit:
fix bug 15152: add databaseId mapping

mind-echo commented:
> LGTM BTW, do we need to do some mapping ? as following :
> 
> ```
> @Bean
> public DatabaseIdProvider databaseIdProvider() {
>     VendorDatabaseIdProvider databaseIdProvider = new VendorDatabaseIdProvider();
>     Properties properties = new Properties();
>     properties.put("Oracle","oracle");
>     properties.put("MySQL","mysql");
>     properties.put("PostgreSQL","pg");
>     databaseIdProvider.setProperties(properties);
>     return databaseIdProvider;
> }
> ```
> 
> cc @EricGao888

Thank you for your review.  I have made changes to the current code

fuchanghai commented:
hi @destroydestiny Is it possible to write an enumeration class to reduce the magic value?

Commit:
fix bug 15152: add the DatabaseId enum to describe the mapping between the productName and the databaseId

mind-echo commented:
> hi @destroydestiny Is it possible to write an enumeration class to reduce the magic value?

I added an enum, but I'm not sure if I added it in the right package

SbloodyS commented on dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/enums/DatabaseId.java:
```diff
@@ -0,0 +1,37 @@
+public enum DatabaseId {
+
+    H2("H2", "h2"),
+    ORACLE("Oracle", "oracle"),
```
We do not support oracle database as meta database.

Commit:
fix bug 15152: remove DatabaseId that are not needed

Commit:
fix bug 15152: optimize code style

ruanwenjun commented on dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/enums/DatabaseId.java:
```diff
@@ -0,0 +1,36 @@
+ */
+@AllArgsConstructor
+@Getter
+public enum DatabaseId {
+
+    H2("H2", "h2"),
+    MYSQL("MySQL", "mysql"),
+    POSTGRESQL("PostgreSQL", "pg");
+
+    private final String productName;
+    private final String databaseId;
+}
```
Can we move this into ado-plugin https://github.com/apache/dolphinscheduler/tree/dev/dolphinscheduler-dao-plugin, and add method to get databaseId mapping at https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-dao-plugin/dolphinscheduler-dao-api/src/main/java/org/apache/dolphinscheduler/dao/plugin/api/DaoPluginConfiguration.java

Commit:
fix bug 15152: Remove DatabaseId enum. Add a method to DaoPluginConfiguration that returns a databaseId to create a FixedDatabaseIdProvider.

ruanwenjun reviewed (dismissed):
LGTM

ruanwenjun commented on dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/DaoConfiguration.java:
```diff
@@ -61,6 +63,11 @@ public MybatisPlusInterceptor paginationInterceptor(DbType dbType) {
 
+    @Bean
+    public DatabaseIdProvider databaseIdProvider() {
+        return new FixedDatabaseIdProvider(daoPluginConfiguration.databaseId());
```
Can we directly use daoPluginConfiguration.dbType().name() as databaseId?
I find the databaseId only exist in `TriggerRelationMapper.xml` if we modify the name to `POSTGRE_SQL` this should work.

mind-echo replied:
that's what I thought at first, use `DatabaseProductName` as databaseId. view commit https://github.com/apache/dolphinscheduler/pull/15153/commits/19d851011656c9fcc4dc5e07241fe98662340b8c

ruanwenjun replied:
If so, we can remove databaseId, and use the default `VendorDatabaseIdProvider` without add properties after change the datasourceId to PostgreSQL in mapper? .

mind-echo replied:
yup

ruanwenjun replied:
Please remove databaseId in DaoPlugin, and use this way.

mind-echo replied:
Can I close this PR and recreate it? Too many commits are not clean enough

ruanwenjun replied:
You can rebase the commit, and I will use squash merge to edit the commit message when merged. So this is not a problem, don't worry.

Commit:
fix bug 15152: use DatabaseProductName as databaseId

ruanwenjun reviewed (approved):
LGTM

SbloodyS reviewed (approved):
+1

ruanwenjun commented:
Merged, thanks for you pr. BTW, we don't only exist UT under h2, are you interesting add test for PG @destroydestiny
